File: app/src/main/java/com/numplates/nomera3/data/network/core/ResponseWrapper.java
package com.numplates.nomera3.data.network.core;

import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * created by c7j on 24.05.18
 */
public class ResponseWrapper<T> implements Serializable {

    public static int CODE_400 = 400;

    @SerializedName("success")
    @Nullable
    private T data;

    @SerializedName("error")
    @Nullable
    private ResponseError err;

    @SerializedName("code")
    private int code;

    @SerializedName("message")
    @Nullable
    private String message;

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public ResponseError getErr() {
        return err;
    }

    public void setErr(ResponseError err) {
        this.err = err;
    }

    public int getCode() { return this.code; }

    public void setCode(int code) { this.code = code; }

    public String getMessage() { return message; }

    public void setMessage(@Nullable String message) { this.message = message; }

    @Override
    public String toString() {
        return "ResponseWrapper{" +
            "data=" + data +
            ", err=" + err +
            ", code=" + code +
            ", message='" + message + '\'' +
            '}';
    }
}

File: core/src/main/java/com/meera/core/network/ResponseWrapper.kt
package com.meera.core.network

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class ResponseWrapper<T> : Serializable {

    @SerializedName("success")
    var data: T? = null
        private set

    @SerializedName("error")
    var err: ResponseError? = null

    @SerializedName("code")
    var code = 0

    @SerializedName("message")
    var message: String? = null

    fun setData(data: T) {
        this.data = data
    }
}

data class ResponseError(
    @SerializedName("code") var code: Int,
    @SerializedName("message") var message: String?,
    @SerializedName("user_message") var userMessage: String?
) : Serializable
